controllers/keystoneapi_controller.go

```diff
@@ -1205,11 +1227,49 @@ func (r *KeystoneAPIReconciler) generateServiceConfigMaps(
 		endptConfig := map[string]interface{}{}
 		endptConfig["ServerName"] = fmt.Sprintf("%s-%s.%s.svc", instance.Name, endpt.String(), instance.Namespace)
 		endptConfig["TLS"] = false // default TLS to false, and set it bellow to true if enabled
+		endptConfig["EnableFederation"] = enableFederation
 		if instance.Spec.TLS.API.Enabled(endpt) {
 			endptConfig["TLS"] = true
 			endptConfig["SSLCertificateFile"] = fmt.Sprintf("/etc/pki/tls/certs/%s.crt", endpt.String())
 			endptConfig["SSLCertificateKeyFile"] = fmt.Sprintf("/etc/pki/tls/private/%s.key", endpt.String())
 		}
+
+		if enableFederation {
+			ospSecret, _, err := oko_secret.GetSecret(
+				ctx,
+				h,
+				instance.Spec.Secret,
+				instance.Namespace)
+			if err != nil {
+				return err
+			}
+
+			OIDCClientSecret := string(ospSecret.Data[instance.Spec.PasswordSelectors.KeystoneOIDCCryptoPassphrase])
+			if OIDCClientSecret == "" {
+				return errors.New("OIDCClientSecret cannot be empty")
```

Comment:
we should probably also enhance L721
```
	hash, result, err := oko_secret.VerifySecret(ctx, types.NamespacedName{Name: instance.Spec.Secret, Namespace: instance.Namespace}, []string{"AdminPassword"}, helper.GetClient(), time.Second*10)
```
to make `[]string{"AdminPassword"}` conditional to check for `KeystoneOIDCCryptoPassphrase` and `KeystoneOIDCClientSecret`. this would also add them to the hash, in case the passwords change. This will not catch if the password is an empty string, but make sure an entry exist in the secret.